### PR TITLE
test(toast): refactor for cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor Cypress tests for `Toast` component to use new cypress-component-react framework for testing.
Add toast.test.js.

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run npx cypress open-ct to check if there is newly added test.file
- [x] Check if the `toast.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/toast/toast.test.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+    it("should render Toast component with focused close icon", () => {
+      CypressMountWithProviders(<ToastComponent />);
+
+      closeIconButton()
+        .should("have.css", "outline-color", "rgb(255, 181, 0)")
+        .and("have.css", "outline-width", "3px");
+    });
+
+    it("should render Toast component with not focused close icon", () => {
```
Comment (stephenogorman) [inline src/components/toast/toast.test.js:66]: it("should render Toast component with unfocused close icon", () => {

Comment (stephenogorman) [inline src/components/toast/toast.test.js:66]: it("should render Toast component with unfocused close icon", () => {

File: src/components/toast/toast.test.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      CypressMountWithProviders(<ToastComponent onDismiss={callback} />);
+
+      closeIconButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+  });
```
Comment (stephenogorman) [inline src/components/toast/toast.test.js:174]: Should the onDismiss callback be tested when Esc is used to close component?

**Review by stephenogorman — CHANGES_REQUESTED** [review]

**Review by sianford — APPROVED** [review]

File: src/components/toast/toast.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/toast/toast.test.js:188]: agree, thx for pointing this, added a TC

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**